internal/models/challenge.go

```diff
@@ -1,20 +1,56 @@
 package models
 
 import (
+	"database/sql/driver"
+	"fmt"
+
+	"encoding/json"
+	"github.com/go-webauthn/webauthn/webauthn"
 	"github.com/gofrs/uuid"
 	"github.com/supabase/auth/internal/crypto"
 	"github.com/supabase/auth/internal/storage"
 	"time"
 )
 
 type Challenge struct {
-	ID         uuid.UUID  `json:"challenge_id" db:"id"`
-	FactorID   uuid.UUID  `json:"factor_id" db:"factor_id"`
-	CreatedAt  time.Time  `json:"created_at" db:"created_at"`
-	VerifiedAt *time.Time `json:"verified_at,omitempty" db:"verified_at"`
-	IPAddress  string     `json:"ip_address" db:"ip_address"`
-	Factor     *Factor    `json:"factor,omitempty" belongs_to:"factor"`
-	OtpCode    string     `json:"otp_code,omitempty" db:"otp_code"`
+	ID                  uuid.UUID            `json:"challenge_id" db:"id"`
+	FactorID            uuid.UUID            `json:"factor_id" db:"factor_id"`
+	CreatedAt           time.Time            `json:"created_at" db:"created_at"`
+	VerifiedAt          *time.Time           `json:"verified_at,omitempty" db:"verified_at"`
+	IPAddress           string               `json:"ip_address" db:"ip_address"`
+	Factor              *Factor              `json:"factor,omitempty" belongs_to:"factor"`
+	OtpCode             string               `json:"otp_code,omitempty" db:"otp_code"`
+	WebAuthnSessionData *WebAuthnSessionData `json:"web_authn_session_data,omitempty" db:"web_authn_session_data"`
+}
+
+type WebAuthnSessionData struct {
+	*webauthn.SessionData
+}
+
+func (s *WebAuthnSessionData) Value() (driver.Value, error) {
+	return json.Marshal(s)
+}
+
+func (s *WebAuthnSessionData) Scan(value interface{}) error {
+	bytes, ok := value.([]byte)
+	if !ok {
+		return fmt.Errorf("expected byte array but got %T", value)
```

Comment:
Is this realistic? I'd rather we `panic` for really unexpected cases like this.

Also can't it sometimes be `string` also?

Reply:
Yes let's panic and also handle for other cases